File: src/utils/homeRefresh.ts
interface HomeRefreshDeps {
  fetchWeather: (lat: number, lng: number, force?: boolean) => Promise<void>;
  fetchTodayStatus: () => Promise<void>;
  fetchFeedbackCount: () => Promise<void>;
  fetchTodayPrediction: () => Promise<void>;
}

interface HomeRefreshParams {
  lat: number;
  lng: number;
  force?: boolean;
}

export async function refreshHomeScreenData(
  deps: HomeRefreshDeps,
  params: HomeRefreshParams,
): Promise<void> {
  const { lat, lng, force = false } = params;

  await Promise.all([
    deps.fetchWeather(lat, lng, force),
    deps.fetchTodayStatus(),
    deps.fetchFeedbackCount(),
  ]);

  await deps.fetchTodayPrediction();
}

File: apps/android-native/app/src/main/java/com/wxxtae/pws/nativepreview/domain/HomeRefresh.kt
package com.wxxtae.pws.nativepreview.domain

data class HomeRefreshParams(
    val lat: Double,
    val lng: Double,
    val force: Boolean = false,
)

interface HomeRefreshDeps {
    fun fetchWeather(lat: Double, lng: Double, force: Boolean = false)
    fun fetchTodayStatus()
    fun fetchFeedbackCount()
    fun fetchTodayPrediction()
}

object HomeRefreshCoordinator {
    /**
     * Mirrors the RN contract: weather/status/feedback are prerequisites,
     * then prediction runs after those inputs are refreshed.
     */
    fun refreshHomeScreenData(deps: HomeRefreshDeps, params: HomeRefreshParams) {
        deps.fetchWeather(params.lat, params.lng, params.force)
        deps.fetchTodayStatus()
        deps.fetchFeedbackCount()
        deps.fetchTodayPrediction()
    }
}
